lua/telescope/utils.lua

```diff
@@ -170,7 +170,63 @@ utils.filter_symbols = function(results, opts, post_filter)
   end
 end
 
-utils.path_smart = (function()
+local path_filename_first = function(path, reverse_directories)
+  local dirs = vim.split(path, utils.get_separator())
+  local filename
+
+  if reverse_directories then
+    dirs = utils.reverse_table(dirs)
+    filename = table.remove(dirs, 1)
+  else
+    filename = table.remove(dirs, #dirs)
+  end
+
+  local tail = table.concat(dirs, utils.get_separator())
+  -- Trim prevents a top-level filename to have a trailing white space
+  local transformed_path = vim.trim(filename .. " " .. tail)
+  local path_style = { { { #filename, #transformed_path }, "TelescopeResultsComment" } }
+
+  return transformed_path, path_style
+end
+
+local calc_result_length = function(truncate_len)
+  local status = get_status(vim.api.nvim_get_current_buf())
+  local len = vim.api.nvim_win_get_width(status.layout.results.winid) - status.picker.selection_caret:len() - 2
+  return type(truncate_len) == "number" and len - truncate_len or len
+end
+
+local path_truncate = function(path, truncate_len, opts)
+  if opts.__length == nil then
+    opts.__length = calc_result_length(truncate_len)
+  end
+  if opts.__prefix == nil then
+    opts.__prefix = 0
+  end
+  return truncate(path, opts.__length - opts.__prefix, nil, -1)
+end
+
+local path_shorten = function(path, length, exclude)
+  if exclude ~= nil then
+    return Path:new(path):shorten(length, exclude)
+  else
+    return Path:new(path):shorten(length)
+  end
+end
+
+local path_abs = function(path, opts)
+  local cwd
+  if opts.cwd then
+    cwd = opts.cwd
+    if not vim.in_fast_event() then
+      cwd = utils.path_expand(opts.cwd)
+    end
+  else
+    cwd = vim.loop.cwd()
+  end
+  return Path:new(path):make_relative(cwd)
+end
+
+local path_smart = (function()
```

Comment:
sadly we cant make path_smart local because it was already exported for months. I was only talking about new function additions. changing this would result in a breaking change which